### DISCUSSION TITLE
Add PNG Frame Around Image

### DISCUSSION THREAD

**Discussion by GoremanX** [ImageMagick/ImageMagick#7255] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10

### Description

I have a directory with about 400 JPG photos that are all exactly the same size (1600x2000). To each photo, I want to add a transparent border that's 182px on the left/right, 304px on top, and 521px on the bottom, overlay a PNG file on top of the whole thing (that's my photo frame), and save the result as a PNG file in a different directory with the same filename.

Help with coming up with a proper command line for this would be hugely appreciated.

**Comment by snibgo** [comment]
So each result will be 182+1600+182 = 1964 pixels wide, and 304+2000+521 = 2825 pixels high. Is that correct? Then I suppose the PNG file for the border is 1964x2825 pixels. Is that correct?

The IM command could be:
```
magick border.png in.jpg -geometry +182+304 -composite otherdir\out.png
```
This would put the JPG inage on top of the larger PNG image. I'm not sure that this is what you want.

To do all the JPGs in the current directory, put that in a `for` loop.

**Comment by GoremanX** [comment]
not quite, but that's close. I want the PNG file overlaid on top of the JPG file, and I want to be sure that the original JPG is placed within the frame opening at the correct location, hence the very specific transparent border dimensions. Here's a specific example, I want to take these hundreds of portraits:

![802M4008-Enhanced-NR](https://github.com/ImageMagick/ImageMagick/assets/876310/5d2ba041-5fc6-4774-92bd-e92651252714)

and overlay this frame over them:

![polaroid-frame copy](https://github.com/ImageMagick/ImageMagick/assets/876310/3048bcf2-97c7-45bb-98e3-ba8a134ab32e)

There's transparency around that frame that I want to maintain in the resulting PNG. That's why I thought I should grow the original file using the correct-size transparent borders and then overlay the PNG frame on top of it. I already worked out the dimensions of all files so that the portraits fit within that frame opening without any need to scale.

**Comment by snibgo** [comment]
How about this:
```
magick portBorder.png portrait.jpg -geometry +182+304 -composite out.png
```

**Comment by GoremanX** [comment]
that's almost perfect! but the JPG is still on top of the PNG, and I want the reverse because the inside transparent area of the PNG has feathered edge that I want to retain

**Comment by snibgo** [comment]
Okay. Perhaps:
```
magick portBorder.png portrait.jpg -geometry +182+304 -compose DstOver -composite out.png
```

**Reply by GoremanX** [reply]
It's PERFECT! Thank you so much, and it was very helpful to see the syntax so I can figure out other functions myself in the future
